junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/LauncherDiscoveryRequestBuilder.java

```diff
@@ -323,6 +324,19 @@ else if (filter instanceof DiscoveryFilter<?> discoveryFilter) {
 		}
 	}
 
+	/**
+	 * Builds this discovery request and returns a new builder for creating a
+	 * {@link org.junit.platform.launcher.LauncherExecutionRequest} that is
+	 * initialized to contain the resulting discovery request.
+	 *
+	 * @return a new {@link LauncherExecutionRequestBuilder}
+	 * @since 6.0
+	 */
+	@API(status = EXPERIMENTAL, since = "6.0")
+	public LauncherExecutionRequestBuilder forExecution() {
```

Comment:
The following would be canonical, but also a mouthful:
```java
	public LauncherExecutionRequestBuilder toLauncherExecutionRequestBuilderExecution() {
```

I think, that `forExecution` is a good-enough abbreviation here.